mobile-browser-based-version/src/components/MNISTComponents/MNIST_script.js

```diff
@@ -0,0 +1,146 @@
+    tradeoffs: "We are using a simple model, first a 2d convolutional layer > max pooling > 2d convolutional layer > max pooling > convolutional layer > 2 dense layers.",
+    // {String} information about expected data 
+    dataFormatInformation: 
+    "This model takes as input an image dataset. You can upload each digit image of your dataset in the box corresponding to its label. The size of the image will be reduced to 28x28. Finally, the image should be in grey scale.",
+    // {String} description of the datapoint given as example
+    dataExampleText: "Below you can find an example of an expected image representing the digit 9.",
+};
+
+/**
+ * Object used to contain the model's training specifications
+ */
+export const training_information = {
+    // {String} model's identification name
+    model_id: "mnist-model",
+    // {Number} port of the peerjs server
+    port: 1,
+    // {Number} number of epoch used for training
+    epoch: 10,
+    // {Number} validation split
+    validation_split: 0.2,
+    // {Number} batchsize
+    batch_size: 30,
+    // {Object} Compiling information 
+    model_compile_data: {
+        optimizer: "rmsprop",
+        loss: "binaryCrossentropy",
+        metrics: ["accuracy"],
+    },
+    // {Object} Training information 
+    model_train_data: {
+        epochs: 10,
+    },
+}
+
+/**
+ * This functions takes as input a file (of type File) uploaded by the reader and checks 
+ * if the said file meets the constraints requirements and if so prepare the training data. 
+ * @param {Dictionary[ImageURL: String]} training_data is a dictionary between the image_url and the label
+ * @returns an object of the form: {accepted: Boolean, Xtrain: training data, ytrain: data's labels}
+ */
+export async function data_preprocessing(training_data) {
+    console.log("Start: Processing Uploaded File");
+    var Xtrain = null;
+    var ytrain = null;
+
+    // Check some basic prop. in the user's uploaded file
+    
+    var check_result= await check_data(training_data)
+    var startTraining = check_result.accepted
+
+    // If user's file respects our format, parse it and start training
+    if (startTraining) {
+        const labels = []
+        const image_uri = []
+
+        Object.keys(training_data).forEach(key => {
+            console.log(key, training_data[key])
+            labels.push(training_data[key])
+            image_uri.push(key)
+        });
+
+        console.log("User File Validated. Start parsing.");
+
+        // Do feature preprocessing
+        ytrain = labels_preprocessing(labels)
+        const image_tensors = []
+    
+        image_uri.forEach( image => 
+        image_tensors.push(image_preprocessing(image))
+        )
+ 
+        Xtrain = tf.concat(image_tensors, 0)
+        // object to return 
+    } else {
+        console.log("Cannot start training.")
+    }
+    return {accepted: startTraining, Xtrain: Xtrain, ytrain: ytrain}
+}
+
+async function check_data(training_data){
```

Comment:
Maybe we can put this function in a helper script for all images script (as it's not task-specific). In the future, we could even add more methods that are image-specific in this script.